src/drivers/intel/cavs/ipc.c

```diff
@@ -219,15 +204,17 @@ void ipc_platform_complete_cmd(void *data)
 	/* unmask Busy interrupt */
 	ipc_write(IPC_DIPCCTL, ipc_read(IPC_DIPCCTL) | IPC_DIPCCTL_IPCTBIE);
 
-#if CONFIG_SUECREEK
+	/* are we about to enter D3 ? */
 	if (ipc->pm_prepare_D3) {
-
+#if CONFIG_SUECREEK
 		//TODO: add support for Icelake
 		while (1)
 			wait_for_interrupt(0);
-	}
-
+#else
+		/* no return - memory will be powered off and IPC sent */
+		platform_pm_runtime_power_off();
 #endif
+	}
```

Comment:
@lyakh this part was specifically written this way by @mmaka1. IIRC correctly sending the host notification first before powering off caused some issues back in the CML days. Thats why we power off the memory and then send the notification to the host. And we ensure not to check the reply for the CTX_SAVE IPC because the memory is already power gated for this case. @mmaka1 can you please confirm if this change is OK?

Reply:
in any case, what we could do is move the power_off to before the ipc_write to keep the same sequence

Reply:
that patch isn't crucial for this series. If it's wrong, we can easily drop it or I can move it out to a separate PR for more thorough consideration. FWIW all tests passed so far, except one, likely unrelated. And if it is indeed wrong, we might want to add a test for this.

Reply:
agree. this patch is not that critical. The CTX_SAVE IPC is the only one that will be power gating the memory and it should never be executing on a secondary core in the first place

Reply:
@ranj063 so what do we do with this? @mmaka1 ? @lgirdwood any ideas?